core/modules/ckeditor5/js/ckeditor5-prepare.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * @file
+ * Backwards compatibility of UMD build to integration via DLL builds.
+ */
+(function () {
+  "use strict";
+
+  // Mimic some of the DLL structure by adding some aliases for commonly
+  // extended classes. It won't be possible to mimic the entire DLL structure.
+  // This must happen before any custom plugin js files load.
+  CKEDITOR.core = {};// Does not exist in UMD.
+  CKEDITOR.core.Plugin = CKEDITOR.Plugin;
+  CKEDITOR.core.Command = CKEDITOR.Command;
+
+  // Icon location and names have changed in CKEditor5 v45. These two icons are
+  // used by core plugins.
+  // There's some risk, that location and names change again at some point.
+  // @see https://github.com/ckeditor/ckeditor5/issues/17779#issuecomment-2882767192
+  //
+  // @todo should we expose all of then in their old location (currently that's
+  // 763 icons)?
+  CKEDITOR.core.icons = {
+    pencil: CKEDITOR.IconPencil,
+    image: CKEDITOR.IconImage
+  }
+
+  CKEDITOR.ui = {};// Does not exist in UMD.
+  CKEDITOR.ui.ButtonView = CKEDITOR.ButtonView;
+
+  // Use a different name to expose to global space - the one we used before.
+  window.CKEditor5 = CKEDITOR;
+  // Prevent conflict with CKE4, depends on order, v4 has to run after this.
+  delete window.CKEDITOR;
```

Comment:
This seemed awkward to me at first, but it makes all behavior consistent with earlier CKEditor 5 versions, so I think this is okay. Having _both_ `CKEDITOR` and `CKEditor5` window globals would likely cause confusion; in addition to the conflict problem with CKEditor 4.